src/Chapter2.hs

```diff
@@ -348,9 +355,10 @@ Implement a function that returns only the first half of a given list.
 >>> firstHalf "bca"
 "b"
 -}
--- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
-
+firstHalf :: [a] -> [a]
+firstHalf [] = []
+firstHalf xs = take (size `div` 2) xs
+    where size = length xs
```

Comment:
Nice usage of the `where` construction 👷🏼 

Note that the first case is included in the general solution as well. It will safe a few function calls though 🤔 👍🏼